fs/base.py

```diff
@@ -1133,6 +1155,7 @@ def move(self, src_path, dst_path, overwrite=False):
             with self.open(src_path, "rb") as read_file:
                 # FIXME(@althonos): typing complains because open return IO
                 self.upload(dst_path, read_file)  # type: ignore
+            copy_mtime(self, src_path, self, dst_path)
```

Comment:
Same here:
```python
if preserve_time:
    copy_mtime(...)
```

Reply:
Also, what happens if both resources have a syspath? There is an early return above, so I don't think the *mtime* is copied in that case.

Reply:
I think `os.rename` doesn't actually touch the mtime value, at least on UNIX and Windows. I added the check anyway, just to be safe.